test/libsolidity/syntaxTests/operators/custom/implicit_conversion_is_blocked.sol

```diff
@@ -0,0 +1,34 @@
+using {add as +, unsub as -} for S;
+
+struct S {
+    uint x;
+}
+
+function add(S memory _a, S memory) returns (S memory) {
+    return _a;
+}
+
+function unsub(S memory _a) returns (S memory) {
+    return _a;
+}
+
+contract C {
+    S s;
+
+    function test() public {
+        S memory sTmp;
+        s + s;
+        sTmp + true;
+        true + s;
+        -sTmp;
+        -s;
+        -true;
+    }
+}
+
+// ----
+// TypeError 2271: (288-293): Operator + not compatible with types struct S storage ref and struct S storage ref. No matching user-defined operator found.
```

Comment:
I think in general we should really improve the readability of this. I think I created once a PR but I did not finish it. The main problem is, that it's hard to see to what line those errors exactly referring to.

Reply:
Ok, found it. @leonardoalt closed it https://github.com/ethereum/solidity/pull/12890. Yeah, I still think that would be nice to have.

Reply:
We only closed it because we're hit 100 PRs and we had to be really aggressive with closing them go get down to a more reasonable number. Feel free to reopen if you want to finish it, just at the moment we're still flooded with PRs...